ios/RNSScreenStack.mm

```diff
@@ -1178,6 +1174,24 @@ - (void)mountingTransactionWillMount:(react::MountingTransaction const &)transac
   }
 }
 
+- (void)mountingTransactionDidMount:(const facebook::react::MountingTransaction &)transaction
+               withSurfaceTelemetry:(const facebook::react::SurfaceTelemetry &)surfaceTelemetry
+{
+  for (auto &mutation : transaction.getMutations()) {
+    if (mutation.parentShadowView.componentName != nil &&
+        strcmp(mutation.parentShadowView.componentName, "RNSScreenStack") == 0 &&
+        (mutation.type == react::ShadowViewMutation::Type::Update ||
+         mutation.type == react::ShadowViewMutation::Type::Remove)) {
+      // we need to wait until children have their layout set. At this point they don't have the layout
+      // set yet, however the layout call is already enqueued on ui thread. Enqueuing update call on the
+      // ui queue will guarantee that the update will run after layout.
+      dispatch_async(dispatch_get_main_queue(), ^{
+        [self maybeAddToParentAndUpdateContainer];
+      });
+    }
```

Comment:
```suggestion
    if (mutation.parentShadowView.componentName != nil &&
        strcmp(mutation.parentShadowView.componentName, "RNSScreenStack") == 0 &&
        (mutation.type == react::ShadowViewMutation::Type::Insert ||
         mutation.type == react::ShadowViewMutation::Type::Remove)) {
      // we need to wait until children have their layout set. At this point they don't have the layout
      // set yet, however the layout call is already enqueued on ui thread. Enqueuing update call on the
      // ui queue will guarantee that the update will run after layout.
      dispatch_async(dispatch_get_main_queue(), ^{
        [self maybeAddToParentAndUpdateContainer];
      });
      break;
    }
```

So the `Update` here worked just by accident, because there was only single `Update` per transaction per Screen (dunno why) & when you put `Insert` here there were two consecutive inserts one by one thus leading to the same crash as calling `updateContainer` in `mountChildComponentView:index:`. I propose using `Insert` here, as that makes much more sense & just breaking after a update is scheduled, as it doesn't seem to make sense to schedule the block multiple (possibly a big number) of times per every transaction.

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/2249/commits/3d89cfd66f04044c2fb3062765f7becd6dbe3f62.